.github/workflows/ci.yml

```diff
@@ -47,12 +47,23 @@ jobs:
         if: ${{ !cancelled() && steps.auto_install_edge.conclusion == 'success' }}
         run: make compose-build nfs-build
 
+      - name: Set GIT_BRANCH
+        id: set_git_branch
+        if: ${{ !cancelled() && steps.build_images.conclusion == 'success' }}
+        run: |
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+            GIT_BRANCH="${{ github.head_ref }}"
+          else
+            GIT_BRANCH="${{ github.ref#refs/heads/ }}"
+          fi
+          echo "GIT_BRANCH=$GIT_BRANCH" >> $GITHUB_ENV
+
       - name: Use auto-install script to upgrade containers to latest version
         id: auto_install_upgrade
-        if: ${{ !cancelled() && steps.build_images.conclusion == 'success' }}
+        if: ${{ !cancelled() && steps.set_git_branch.conclusion == 'success' }}
```

Comment:
Are you sure of this change? What happens if that step is not executed because `steps.build_images.conclusion` fails?

Reply:
Then, the "Set GIT_BRANCH" step would not execute 

https://github.com/openwisp/docker-openwisp/pull/331/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR52